### DISCUSSION TITLE
Technitium Container on MikroTik

### DISCUSSION THREAD

**Discussion by joshg207** [TechnitiumSoftware/DnsServer#1115] [Q&A] [answered]
I've searched the discussions and issues, and haven't found many instances of people indicating they have run Technitium container on Mikrotik hardware. Mostly just looking for anyone's experiences with doing so. [I did find one mention](https://github.com/TechnitiumSoftware/DnsServer/issues/852).

I frequently lose power, and have most of my hardware shutdown gracefully when this happens. One downside is that I lose for core infrastructure, like DNS. However, I already leave my router, and switches running, since they consume less power, and this prevents the wait for them to come back online when its just a momentary blip. Since routerOS can host containers, and its own implementation of DNS leaves much to be desired, it seems like a decent match. [There are security implications](https://help.mikrotik.com/docs/spaces/ROS/pages/84901929/Container#Container-Disclaimer), as highlighted by MikroTiks own docs.

At first consideration, it seems attractive to have my DNS reside within the router/switches, keeping a majority of networking related processes within specific hardware. Freeing me up to adjust hardware on my prox/esxi hosts without having to worry about where networking related services are running. RouterOS backups should cover these as well from what I've read. At the moment, all of my DNS zones, replication, etc... are handled through old perl scripts manually. I've briefly experimented with pihole, and have decent experiance with freeIPA DNS (the limited feature set it has), but not any other web/GUI based DNS administration tools.

So, if anyone has experience doing this or something similar, I'd love to know more. I've seen[ various repos of known working containers](https://github.com/akmalovaa/mikrotik-containers), like pihole, nginx etc... on MikroTik hardware. I was going to probably give it a shot this coming weekend, but my partner is pretty over the "Hey, internet is going down for a minute," and me inevitably rolling back to the original configuration 5 hours later, so just doing some asking around first.

FWIW, my primary network hardware is.
Router: [RB5009](https://mikrotik.com/product/rb5009ug_s_in) - ARM64
1GbE Switch: [CRS326](https://mikrotik.com/product/CRS326-24G-2SplusRM) - ARMX86
10GbE Switch: [crs312](https://mikrotik.com/product/crs312_4c_8xg_rm) - QCA9531 The architecture of this switch means no containers, I'm fairly certain.

**Comment by ShreyasZare** [comment, marked as answer]
Thanks for the post. I have had some queries from people running it on MikroTik with docker. But, I don't have much info regarding the setup. I would only suggest to not use block lists since the router may have less memory available and all block lists are loaded in memory by the DNS server and thus may consume more RAM if large lists are configured.

I would suggest that you post this on reddit.com/r/technitium to get more feedback from people who are using it.

**Reply by micmejia** [reply]
I'm setting up right now to use technitium dns on a mikrotik router (rb5009, with 1GB RAM), same with OP. I plan to use the block list, which currently has around 200K+ blocked urls. Do you have an estimate how much RAM will this consume?

```
[2025-03-07 00:15:33 UTC] DNS Server successfully downloaded block list (5.61 MB): https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/porn/hosts
[2025-03-07 00:15:33 UTC] DNS Server is reading block list from: https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/porn/hosts
[2025-03-07 00:15:34 UTC] DNS Server read block list file (204055 domains) from: https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/porn/hosts
[2025-03-07 00:15:34 UTC] DNS Server block list zone was loaded successfully.
[2025-03-07 00:15:44 UTC] DNS Server config file was saved: /etc/dns/dns.config
```

With the block list loaded, and the router is on idle (no requests handled at the moment during setup), it has about 595 MB of ram available/left. Should I avoid using the block list in this kind of setup? Thanks.

**Reply by ShreyasZare** [reply]
@micmejia Thanks for asking. It really depends on how much RAM is free and how much it is usually needed by the router itself. So, you can try it for a couple of days and if its working well without any crash then there should not be any issues.
